src/model.ts

```diff
@@ -963,7 +963,7 @@ class Model<T> extends EventEmitter {
 
 Model.prototype.get = Model.prototype.findById;
 
-function execHooks(schema: Schema, type: string, event: string, data: any): Promise<any> {
+function execHooks(schema: Schema<any>, type: string, event: string, data: any): Promise<any> {
```

Comment:
```suggestion
function execHooks<T>(schema: Schema<T>, type: string, event: string, data: any): Promise<any> {
```

Let's use a generic here, and TypeScript will be able to infer `T`.